### PR TITLE
fix(leaderelection): Validate and default lease refresh intervals to prevent ticker panics

### PR DESCRIPTION
**Pull request by pranjalj20014**
## Problem
The `leader_election.go` file could hard-crash startup if either `LeaderLeaseRefreshInterval` or `FollowerLeaseRefreshInterval` configuration values were set to zero or negative values. The `time.NewTicker()` function rejects non-positive durations and will panic, but there was no validation before the configuration values were used to create tickers.

This issue occurred at two locations:
- Line 79: `ticker := time.NewTicker(p.acquireLock())`
- Line 105: Inside the loop when creating new ticker instances

## Solution
Implemented **Option 2**: Set sensible default values if the configuration provides zero or negative durations. This provides graceful degradation and prevents application crashes.

### Changes Made:
1. **Added validation in `NewElectionParticipant()` constructor** with defaults:
   - `LeaderLeaseRefreshInterval`: defaults to 5 seconds if ≤ 0
   - `FollowerLeaseRefreshInterval`: defaults to 10 seconds if ≤ 0

2. **Added comprehensive test coverage** (`TestNewElectionParticipantWithInvalidDurations`) that verifies:
   - Zero leader and follower intervals are replaced with defaults
   - Negative intervals are replaced with defaults
   - Valid positive intervals remain unchanged
   - All five test cases pass successfully

### Testing
- All existing tests pass (6 test cases in `TestAcquireLock`)
- New test suite with 5 subtests covers edge cases
- No interference with other modules

## Impact
This is a **defensive fix** that prevents application crashes from misconfiguration while maintaining backward compatibility with valid configurations.

### CODE REVIEW

File: internal/leaderelection/leader_election.go [hunk 1]
```diff
@@ -45,6 +45,14 @@ type ElectionParticipantOptions struct {
 
 // NewElectionParticipant returns a ElectionParticipant which attempts to become leader.
 func NewElectionParticipant(lock dl.Lock, resourceName string, options ElectionParticipantOptions) *DistributedElectionParticipant {
+	// Set sensible defaults if durations are zero or negative
+	if options.LeaderLeaseRefreshInterval <= 0 {
+		options.LeaderLeaseRefreshInterval = 5 * time.Second
+	}
+	if options.FollowerLeaseRefreshInterval <= 0 {
+		options.FollowerLeaseRefreshInterval = 10 * time.Second
+	}
```
[the remark this answers is not in the document]
Comment (pranjalj20014) [reply internal/leaderelection/leader_election.go]: @copilot apply changes based on [this feedback](https://github.com/jaegertracing/jaeger/pull/8589#discussion_r3265672916)
